src/main/msp/msp.c

```diff
@@ -1837,11 +1870,42 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
     case MSP_SET_RESET_CURR_PID:
         resetPidProfile(currentPidProfile);
         break;
-    case MSP_SET_SENSOR_ALIGNMENT:
-        gyroConfigMutable()->gyro_align = sbufReadU8(src);
-        accelerometerConfigMutable()->acc_align = sbufReadU8(src);
+    case MSP_SET_SENSOR_ALIGNMENT: {
+        // maintain backwards compatibility for API < 1.41
+        const uint8_t gyroAlignment = sbufReadU8(src);
+        sbufReadU8(src);  // discard deprecated acc_align
+#ifdef USE_MULTI_GYRO
+        switch (gyroConfig()->gyro_to_use) {
+        case GYRO_CONFIG_USE_GYRO_2:
+            gyroDeviceConfigMutable(1)->align = gyroAlignment;
+            break;
+        case GYRO_CONFIG_USE_GYRO_BOTH:
```

Comment:
I would probably defer to always reading / writing `gyroDeviceConfigMutable(0)->align` when `GYRO_CONFIG_USE_GYRO_BOTH` is configured. At least this way the behaviour in configurator will be transitive (user sets setting, saves -> new setting is shown). Users with an old configurator will only have to adjust `gyroDeviceConfigMutable(1)->align` in CLI then.

Reply:
The thing is though that there is only one board (SPRACINGF7DUAL) that can used mode `GYRO_CONFIG_USE_GYRO_BOTH` and its gyros are `CW0` for gyro 0 and `CW90` for gryo 1. So if we allow the configurator to update the alignments then no matter what one will be wrong. Even saving from the Configuration tab without changing the setting will corrupt the alignment. So the approach that's there is prevent the MSP from updating the config because it'll always be wrong.

Reply:
If we just always only update gyro 0 this setting won't be wrong - gyro 1 will still need to be set appropriately in CLI, but there's no way around that with the already released configurator.

What this pull request does when `GYRO_CONFIG_USE_GYRO_BOTH` is selected is to allow the user to select a value, then save, and then simply show the user that his setting has been ignored. This is bad UX for something that should actually work, and will probably lead to issues for 'configurator is broken', for something that should work.

Reply:
Ok, I can change. I originally had it this way but found it very confusing. The reason is that when in `GYRO_CONFIG_USE_GYRO_BOTH` mode the sensor readings are averaged - both gyro and ACC. So this produces very strange results when viewing the model movement on the Setup tab if one gyro is misaligned and the other is correct. So the user can get into a very confusing state if changing the setting in the Configuration tab and clicking save. The movement on the Setup tab isn't obviously an alignment problem because of the combination of the two readings. I thought that simply not changing the (presumably) consistent existing settings while not ideal, would be the lesser of two evils.

Reply:
Admittedly changing only gyro 0 isn't ideal in the case when both are used - but at least this provides some functionality (change is effective, and reflected when the setting is read back). There simply isn't a way to make this work satisfactorily with an old configurator.
Also n.b. that setting the alignment to default for both gyros as you had it would have resulted in the same model behaviour.

Reply:
Well I wasn't actually setting the gyros to default. I was just returning `DEFAULT` in `MSP_SENSOR_ALIGNMENT ` so the configurator would display it but in `MSP_SET_SENSOR_ALIGNMENT` it wasn't updating the gyro alignments for the "BOTH" case. So it was all for show.